QWeb/internal/javascript.py

```diff
@@ -163,9 +163,11 @@ def get_by_attributes(elements: list[WebElement], locator: str,
                for (var j = 0; j < attrs.length; j++) {
                     if (attrs[j].value.trim() == locator) {
```

Comment:
I think the original idea has been that since we try to find full text, we trim extra spaces from html (there have been multiple cases where this has been a problem). 

Could be worth trimming locator as well, but it's not that easy or common to have leading & trailing spaces in rf.

Reply:
One scenario comes to mind where matching trimming would be necessary:

- get attribute (with leading/trailing ws) from element 
- do something
- verify no item with the retrieved attribute exists to confirm action

If the trim does not match for attr-value and attr-locator then this kind of sequence will misbehave, verify no item would always pass even if the element would be present.

In order to make it work one would need to know that only the attr-value is trimmed and that the attr-locator should be trimmed by user before passing it to the function/kw.

This shouldn't be that common and might not be worth changing though.